File: java_project/src/main/resources/static/js/entities/account.js
class Account {
    constructor(key, userName, password, isActive, email, phoneNumber, avatarName) {
        this.key = key;
        this.userName = userName;
        this.password = password;
        this.isActive = isActive;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.avatarName = avatarName
    }

    validateEmail() {
        return String(this.email).toLowerCase().match(
            /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
        );
    }
}

File: java_project/src/main/java/com/example/java_project/Entities/Account.java
package com.example.java_project.Entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.util.HashSet;
import java.util.Set;

@Setter
@Getter
@Entity
@EntityListeners(AuditingEntityListener.class)
@Table(name = "account")
public class Account extends AbstractEntity {
    @Getter
    @Setter
    private String key;

    @Getter
    @Setter
    @Column(unique = true)
    private String userName;

    @Getter
    @Setter
    private String password;

    @Getter
    @Setter
    private boolean isActive;

    @Getter
    @Setter
    @Column(unique = true)
    private String email;

    @Getter
    @Setter
    private String phoneNumber;

    @Getter
    @Setter
    private String avatarName;

    @ManyToMany
    @JoinTable(name = "user_role", joinColumns = @JoinColumn(name = "user_id"),
            inverseJoinColumns = @JoinColumn(name = "role_id"))
    private Set<Role> roles = new HashSet<>();
}
